src/middleware/passport-auth.ts

```diff
@@ -0,0 +1,168 @@
+import { createPrivateKey } from 'node:crypto';
+
+import passport from 'passport';
+import { Issuer, Strategy as OpenIdStrategy, TokenSet, UserinfoResponse } from 'openid-client';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+import { Strategy as JWTStrategy, ExtractJwt } from 'passport-jwt';
+import { Repository } from 'typeorm';
+
+import { logger } from '../utils/logger';
+import { User } from '../entities/user';
+
+const readPrivateKey = (privateKey: string) => {
+    return createPrivateKey({ key: Buffer.from(privateKey, 'base64'), type: 'pkcs8', format: 'der' });
+};
+
+export const initPassport = async (userRepository: Repository<User>): Promise<void> => {
+    passport.use(
+        'jwt',
+        new JWTStrategy(
+            {
+                jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+                secretOrKey: process.env.JWT_SECRET!
+            },
+            async (jwtPayload, done) => {
+                logger.info('request authenticated with JWT');
+                const user = await userRepository.findOneBy({ id: jwtPayload?.user?.id });
+                return done(null, user);
+            }
+        )
+    );
+
+    if (process.env.AUTH_PROVIDERS?.includes('onelogin')) {
```

Comment:
We can configure which auth providers get enabled via env config. This is useful for turning them off in CI, or if we only want to allow Google in dev for example.

Reply:
This is good... I like configuration to turn things off and on.  We might in production want or need to remove Marvell Google for exmaple or add social logins in the future so its good to be able to turn these things off and on.  Can you add this to the README?